messaging/writer_test.go

```diff
@@ -26,41 +27,67 @@ func TestNewWriter(t *testing.T) {
 func TestMissyWriter_WriteSuccess(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	brokerWriterMock := NewMockBrokerWriter(mockCtrl)
-	msg := &Message{Key: []byte("key"), Value: []byte("value")}
+	msg := &Message{Key: []byte("key"), Value: []byte("value"), Time: tm.Unix(0, 0).UTC()}
 
 	brokerWriterMock.EXPECT().WriteMessages(gomock.Any(), *msg).Return(nil)
 	brokerWriterMock.EXPECT().Close().Return(nil)
 
 	writer := missyWriter{brokerWriter: brokerWriterMock}
 
+	exec := false
+
+	monkey.Patch(tm.Now, func() tm.Time {
```

Comment:
This is cool, but. You should patch the brokerWriter `WriteMessage` and inside compare if all data is as expected. Why? because now anyone can use time.Now inside the function `Write` and not set it to the message. Your test will pass, but code will be broken. We need to be sure that time is set and is what we expected.
Btw, after that, this expected flat is redundant. You will check if the call happened bacause you checked message fields.